python/tests/variant_tests/dtype_and_variant.py

```diff
@@ -0,0 +1,369 @@
+                      CAST(tiny_int AS VARIANT) AS tiny_int_varnt,
+                      CAST(small_int AS VARIANT) AS small_int_varnt,
+                      CAST(int AS VARIANT) AS int_varnt,
+                      CAST(bigint AS VARIANT) AS bigint_varnt
+                      FROM varnt_int_tbl"""
+
+
+class varnttst_variant_to_int(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [{"tiny_int": 20, "small_int": 24, "int": 21, "bigint": 23}]
+        self.sql = """CREATE MATERIALIZED VIEW variant_to_int AS SELECT
+                      CAST(tiny_int_varnt AS TINYINT) AS tiny_int,
+                      CAST(small_int_varnt AS SMALLINT) AS small_int,
+                      CAST(int_varnt AS INT) AS int,
+                      CAST(bigint_varnt AS BIGINT) AS bigint
+                      FROM int_to_variant"""
+
+
+# BOOLEAN
+class varnttst_bool_tbl(TstTable):
+    """Define the table used by the bool tests"""
+
+    def __init__(self):
+        self.sql = """CREATE TABLE varnt_bool_tbl(
+                      bool1 BOOL,
+                      bool2 BOOL
+                      )"""
+        self.data = [{"bool1": True, "bool2": False}]
+
+
+class varnttst_bool_to_variant(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [{"bool1_varnt": "true", "bool2_varnt": "false"}]
+        self.sql = """CREATE MATERIALIZED VIEW bool_to_variant AS SELECT
+                      CAST(bool1 AS VARIANT) AS bool1_varnt,
+                      CAST(bool2 AS VARIANT) AS bool2_varnt
+                      FROM varnt_bool_tbl"""
+
+
+class varnttst_variant_to_bool(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [{"bool1": True, "bool2": False}]
+        self.sql = """CREATE MATERIALIZED VIEW variant_to_bool AS SELECT
+                      CAST(bool1_varnt AS BOOL) AS bool1,
+                      CAST(bool2_varnt AS BOOL) AS bool2
+                      FROM bool_to_variant"""
+
+
+# DECIMAL
+class varnttst_decimal_tbl(TstTable):
+    """Define the table used by the decimal tests"""
+
+    def __init__(self):
+        self.sql = """CREATE TABLE varnt_decimal_tbl(
+                      decimall DECIMAL(6, 2)
+                      )"""
+        self.data = [{"decimall": 1111.52}]
+
+
+class varnttst_decimal_to_variant(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [{"decimall_varnt": "1111.52"}]
+        self.sql = """CREATE MATERIALIZED VIEW decimal_to_variant AS SELECT
+                      CAST(decimall AS VARIANT) AS decimall_varnt
+                      FROM varnt_decimal_tbl"""
+
+
+class varnttst_variant_to_decimal(TstView):
+    def __init__(self):
+        # checked manually
+        self.data = [{"decimall": Decimal("1111.52")}]
+        self.sql = """CREATE MATERIALIZED VIEW variant_to_decimal AS SELECT
+                      CAST(decimall_varnt AS DECIMAL(6,2)) AS decimall
```

Comment:
here the result type does not need to match the original type exactly (precision, scale)